build.gradle.kts

```diff
@@ -0,0 +1,156 @@
+    id("jacoco") // Code coverage
+    id("maven-publish")
+    alias(libs.plugins.gradleNexusPublishPlugin)
+}
+
+group = "com.redhat.devtools.intellij"
+version = providers.gradleProperty("projectVersion").get() // Plugin version
+val platformVersion = providers.gradleProperty("ideaVersion").get()
+
+// Set the JVM language level used to build the project.
+java {
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
+    withSourcesJar()
+    withJavadocJar()
+}
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    intellijPlatform {
+        defaultRepositories()
+    }
+}
+
+dependencies {
+    intellijPlatform {
+        create(IntelliJPlatformType.IntellijIdeaCommunity, platformVersion)
+
+        // Bundled Plugin Dependencies. Uses `platformBundledPlugins` property from the gradle.properties file for bundled IntelliJ Platform plugins.
+        bundledPlugins(providers.gradleProperty("platformBundledPlugins").map { it.split(',') })
+
+        // Plugin Dependencies. Uses `platformPlugins` property from the gradle.properties file for plugin from JetBrains Marketplace.
+        plugins(providers.gradleProperty("platformPlugins").map { it.split(',') })
+
+        // for local plugin -> https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin-faq.html#how-to-add-a-dependency-on-a-plugin-available-in-the-file-system
+        //plugins.set(listOf(file("/path/to/plugin/")))
+
+        instrumentationTools()
+
+        testFramework(TestFrameworkType.Platform)
+    }
+
+    implementation(libs.kubernetes.client)
+    implementation(libs.openshift.client)
+    implementation(libs.kubernetes.httpclient.okhttp)
+    implementation(libs.jackson.core)
+    implementation(libs.commons.exec)
+    implementation(libs.commons.lang3)
+    implementation(libs.common.lang)
+
+    // for unit tests
+    testImplementation(libs.junit)
+    testImplementation(libs.assertj.core)
+    testImplementation(libs.mockito.inline)
+    testImplementation(libs.opentest4j) // known issue: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin-faq.html#missing-opentest4j-dependency-in-test-framework
+
+}
+
+intellijPlatform {
+    buildSearchableOptions = false
+}
+
+tasks {
+    wrapper {
+        gradleVersion = providers.gradleProperty("gradleVersion").get()
+    }
+
+    test {
+        useJUnit()
+        systemProperty("tools.dl.path", temporaryDir)
+        jvmArgs("-Djava.awt.headless=true")
+    }
+
+    withType<Test> {
+        configure<JacocoTaskExtension> {
+            isIncludeNoLocationClasses = true
+            excludes = listOf("jdk.internal.*")
+        }
+    }
+
+    register("packageTests", Jar::class) {
+        from(sourceSets.test.get().output)
+        archiveClassifier.set("test")
+    }
+
+    withType<GenerateModuleMetadata> {
+        enabled = false
+    }
+
+    // infamous hack, see https://github.com/gradle-nexus/publish-plugin/issues/354
```

Comment:
for now this is the only solution i found to make the publish pass. 
if buildSearchableOptions is set to true, build fails because there is no plugin.xml
if buildSearchableOPtions is set to false, thepubish task fails with  this file not found... 
FYI, there is no such jar uploaded to nexus with older versions